pkg/instancetype/instancetype.go

```diff
@@ -42,6 +42,7 @@ const (
 )
 
 type Methods interface {
+	Upgrader
```

Comment:
Any reason to keep this as a separate interface?

Reply:
There's no functional reason to have it defined as an interface no. I was just playing around with organising the code a little differently, keeping this in a separate file etc and thought the interface at the top of the `upgrade.go` file made it easier for readers to follow instead of having to refer back to this `Methods` interface.

I'm honestly thinking about breaking `instancetype.go` up in a similar way FWIW, having seperate interfaces and files for the different functionality exposed through `Methods`. `Find`, `Apply`, `Infer` etc.